src/service/mobile_config.proto

```diff
@@ -0,0 +1,129 @@
+syntax = "proto3";
+
+package helium.mobile_config;
+
+// ------------------------------------------------------------------
+// Message Definitions
+// ------------------------------------------------------------------
+
+// == Field Notes ==
+//
+// - Every message including a `signature` field will need to be signed over the
+//   complete message, with the `signature` field set to an empty value.
+//
+// - Every `timestamp` is in milliseconds since unix epoch
+//
+// - Keypair fields are binary encoded public keys, Rust encoding example here:
+//    https://github.com/helium/helium-crypto-rs/blob/main/src/public_key.rs#L347-L354
+
+message hotspot_metadata {
+  // The public key binary address and on-chain identity of the hotspot
+  bytes address = 1;
+  // The res12 h3 index asserted address of the hotspot as a string
+  // where an unasserted hotspot returns an empty string
+  string location = 2;
+}
+
+message hotspot_metadata_req_v1 {
+  // The public key address of the hotspot to look up
+  bytes address = 1;
+  bytes signature = 2;
+}
+
+message hotspot_metadata_res_v1 {
+  hotspot_metadata metadata = 1;
+  uint64 timestamp = 2;
+  bytes signature = 3;
+}
+
+message hotspot_metadata_stream_req_v1 {
+  uint32 batch_size = 1;
+  bytes signature = 2;
+}
+
+message hotspot_metadata_stream_res_v1 {
+  // a list of hotspot metadata numbering up to the request batch size
+  repeated hotspot_metadata hotspots = 1;
+  uint64 timestamp = 2;
+  bytes signature = 3;
+}
+
+message router_get_req_v1 {
+  // Pubkey binary of the requested router
+  bytes pubkey = 1;
+  // Signature over the request by the requesting oracle
+  bytes signature = 2;
+}
+
+message router_get_res_v1 {
+  // Pubkey binary of the requested router, confirming registration
+  bytes pubkey = 1;
+  uint64 timestamp = 2;
+  // Signature over the response by the config service
+  bytes signature = 3;
+}
+
+message router_list_req_v1 {
+  // Signature over the request by the requesting oracle
+  bytes signature = 1;
+}
+
+message router_list_res_v1 {
+  // List of public key binaries of all registered routers
+  repeated bytes routers = 1;
+  uint64 timestamp = 2;
+  // Signature over the response by the config service
+  bytes signature = 3;
+}
+
+message admin_add_key_req_v1 {
+  enum key_type_v1 {
+    // administrative operator key
+    administrator = 0;
```

Comment:
hmm doesnt field numbering start at 1 ?

Reply:
this is an enum embedded within the message which we've tended to do with really specific enums that don't need to be reused across multiple message types

Reply:
ah yes its a enum, ignore